### PR TITLE
Take upstream runner dsym changes

### PR DESCRIPTION
**Pull request by keith**
Seems that the goal of these changes is to get macOS to discover the dsyms when running directly through bazel

### CODE REVIEW

File: apple/internal/templates/apple_simulator.template.py [hunk 1]
```diff
@@ -444,37 +454,71 @@ def temporary_simulator(
...
 
 
+def register_dsyms(dsyms_dir: str):
+  """Adds all dSYMs in `dsyms_dir` to the symbolscache.
+
+  Args:
+    dsyms_dir: Path to directory potentially containing dSYMs
+  """
+  symbolscache_command = [
+      "/usr/bin/symbolscache",
```
Comment (brentleyjones) [inline apple/internal/templates/apple_simulator.template.py:486]: Interesting. We might want to do something like this in rules_xcodeproj.

**Review by brentleyjones — APPROVED** [review]

File: apple/internal/templates/apple_simulator.template.py (hunk 1 above)
Comment (keith) [reply apple/internal/templates/apple_simulator.template.py:486]: theoretically shouldn't be needed, but some folks at wwdc also recommended this to me when instruments didn't symbolicate things

File: apple/internal/templates/apple_simulator.template.py [hunk 2]
```diff
@@ -444,37 +454,71 @@ def temporary_simulator(
...
+      "delete",
+      "--tag",
+      "Bazel",
+      "compact",
+      "add",
+      "--tag",
+      "Bazel",
+  ] + [
+      a
+      for a in pathlib.Path(dsyms_dir).glob(
```
Comment (mattrobmattrob) [inline apple/internal/templates/apple_simulator.template.py:496]: Is this glob ever going to be empty?

**Review by mattrobmattrob — APPROVED** [review]

File: apple/internal/templates/apple_simulator.template.py (hunk 2 above)
Comment (keith) [reply apple/internal/templates/apple_simulator.template.py:496]: yea if your build command doesn't generate dsyms i think it will be

Comment (keith) [reply apple/internal/templates/apple_simulator.template.py:496]: when it's empty symbolscache just deletes in that case:

```
2023-08-25 11:36:31.979 INFO Running command: ['/usr/bin/symbolscache', 'delete', '--tag', 'Bazel', 'compact', 'add', '--tag', 'Bazel']
2023-08-25 11:36:32.119 INFO symbolscache output:   No entries deleted
```

**Pull request merged**